src/simulationMachine.tsx

```diff
@@ -217,8 +219,13 @@ export const simulationMachine = simModel.createMachine(
                   }
                 }
               } else if (event.type === 'RESET') {
-                rootService?.stop();
-                locallyInterpret(rootMachine);
+                stopServices();
+                machines.forEach((machine) => {
+                  locallyInterpret(machine);
+                });
+              } else if (event.type === 'STOP') {
+                stopServices();
+                machines.clear();
```

Comment:
This could be moved into the block responsible for handling the `INTERPRET` machine - we probably want to always call this logic then, so I'm not sure if lifting this up as an additional command that can be dispatched by the parent brings much to the table. And when analyzing this part of the code having them grouped would make it slightly easier to follow.